internal/registry/registry.go

```diff
@@ -442,6 +445,16 @@ func (r *Registry) RegisterOrUpdate(ctx context.Context, providerID peer.ID, add
 		}
 	}
 
+	if len(addrs) != 0 {
+		maddrs, err := stringsToMultiaddrs(addrs)
+		if err != nil {
+			panic(err)
```

Comment:
the addrs are supplied by the provider - it's presumably a failure to ingest but not a panic if they're not valid?

Reply:
If an invalid multiaddress got this far, then it must be a programming error. However, anyone who sees that would want to re-validate that is the case, so yeah, better to handle it as an error.